research/delf/delf/python/training/download_dataset.sh

```diff
@@ -0,0 +1,156 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+# This script dowloands the Google Landmarks v2 dataset. GLDv2 datase image files by split:
+# - TRAIN: 500
+# - INDEX: 100
+# - TEST: 20
+
+image_files_train=$1 # Number of image files to download from the TRAIN split
+image_files_index=$2 # Number of image files to download from the INDEX split
+image_files_test=$3 # Number of image files to download from the TEST split
+
+splits=("train" "test" "index")
+dataset_root_folder=gldv2_dataset
+
+metadata_url="https://s3.amazonaws.com/google-landmark/metadata"
+ground_truth_url="https://s3.amazonaws.com/google-landmark/ground_truth"
+csv_train=(${metadata_url}/train.csv ${metadata_url}/train_clean.csv ${metadata_url}/train_attribution.csv ${metadata_url}/train_label_to_category.csv)
+csv_index=(${metadata_url}/index.csv ${metadata_url}/index_image_to_landmark.csv ${metadata_url}/index_label_to_category.csv)
+csv_test=(${metadata_url}/test.csv ${ground_truth_url}/recognition_solution_v2.1.csv ${ground_truth_url}/retrieval_solution_v2.1.csv)
+
+images_tar_file_base_url="https://s3.amazonaws.com/google-landmark"
+images_md5_file_base_url="https://s3.amazonaws.com/google-landmark/md5sum"
+num_processes=6
+
+make_folder() {
+    # Creates a folder and checks if it exists. Exits if folder creation fails.
+    local folder=$1
+    if [ -d "${folder}" ]; then
+        echo "Folder ${folder} already exists. Skipping folder creation."
+    else
+        echo "Creating folder ${folder}."
+        if mkdir ${folder}; then
+            echo "Successfully created folder ${folder}."
+        else
+            echo "Failed to create folder ${folder}. Exiting."
+            exit 1
+        fi
+    fi
+}
+
+download_file() {
+    # Downloads a file from an URL into a specified folder.
+    local file_url=$1
+    local folder=$2
+    local file_path="${folder}/`basename ${file_url}`"
+    echo "Downloading file ${file_url} to folder ${folder}."
+    pushd . > /dev/null
+    cd ${folder}
+    curl -Os ${file_url}
+    popd > /dev/null
+}
+
+validate_md5_checksum() {
+    # Validate the MD5 checksum of a downloaded file.
+    local content_file=$1
+    local md5_file=$2
+    echo "Checking MD5 checksum of file ${content_file} against ${md5_file}"
+    if [[ "${OSTYPE}" == "linux-gnu" ]]; then
+        content_md5=`md5sum ${content_file}`
+    elif [[ "${OSTYPE}" == "darwin"* ]]; then
+        content_md5=`md5 -r "${content_file}"`
+    fi
+    content_md5=`cut -d' ' -f1<<<"${content_md5}"`
+    expected_md5=`cut -d' ' -f1<<<cat "${md5_file}"`
+    if [[ "$content_md5" != "" && "$content_md5" = "$expected_md5" ]]; then
+        echo "Check passed."
+    else
+        echo "Check failed. MD5 checksums don't match."
```

Comment:
should we exit with error here, in case the check does not pass?

Reply:
You're right. I was hesitant about this since the initial script doesn't exit upon MD5 checksum failure, so I reproduced the same behavior. But it makes sense to exit when hitting an error. Changed the behavior to exiting with an error code.